snuba/migrations/operations.py

```diff
@@ -48,3 +51,34 @@ def __init__(self, storage_set: StorageSetKey, table_name: str) -> None:
 
     def format_sql(self) -> str:
         return f"DROP TABLE IF EXISTS {self.table_name};"
+
+
+class TableEngine:
+    pass
+
+
+class ReplacingMergeTree(TableEngine):
+    def __init__(self, version_column: str):
+        self.__version_column = version_column
+
+    def __str__(self) -> str:
+        return f"ReplacingMergeTree({self.__version_column})"
+
+
+class CreateTable(SqlOperation):
+    def __init__(
+        self,
+        storage_set: StorageSetKey,
+        table_name: str,
+        columns: Sequence[Column],
+        engine: TableEngine,
+    ):
+        self.__table_name = table_name
+        self.__columns = columns
+        self.__engine = engine
+        super().__init__(storage_set)
+
+    def format_sql(self) -> str:
+        columns = ", ".join([col.for_schema() for col in self.__columns])
+
+        return f"CREATE TABLE IF NOT EXISTS {self.__table_name} ({columns} ENGINE {self.__engine});"
```

Comment:
I wonder how hard it would be to just reuse `TableSchema` we have in https://github.com/getsentry/snuba/blob/master/snuba/datasets/schemas/tables.py#L48

I'd rather we avoided building two diverging system to run DDL operations.
Assuming there will be migrations that will create Snuba datasets tables, how do you envision them to work ? Is the migration code going to leverage the current TableSchema structure (from schemas/tables.py) or will it work the other way around ?

Reply:
I really wanted to avoid just trying to fit this into what already exists. I'd rather remove all the DDL parts of table schema later, however they'll have to coexist for a while.

Reply:
Does this mean the DDL will be only in in the migration and not in the schema classes ?

Reply:
Yes, that is the plan. Once the new migrations are being used there shouldn't be any reason for the existing methods that generate DDL and I think we can simplify the schema classes. (Well, apart from the automatic migration behavior that happens when you hit any dataset now but I think we should reconsider that and explicitly run the migrations instead in those situations.)

Reply:
Mind adding a TODO to explain that we are not expecting to have two separate DDL structures?

Reply:
Sure